src/gui/parameter/ParameterWidget.cc

```diff
@@ -106,11 +109,11 @@ void ParameterWidget::saveFile(const QString& scadFile)
   }
 
   QString jsonFile = getJsonFile(scadFile);
-  if (jsonFile == this->invalidJsonFile) {
+  if (this->jsonError.hasError() && jsonFile.toStdString() == this->jsonError.filename) {
```

Comment:
What does the second check do? Are there situations where these filenames may differ?